pkg/platform/coverage/golang/golang.go

```diff
@@ -62,8 +62,27 @@ func (g *Golang) GetCoverage() (models.TestCoverage, error) {
 		TotalCov: "",
 	}
 
+	coverageDir := os.Getenv("GOCOVERDIR")
```

Comment:
what is this code about..?

Reply:
Earlier, if coverage files are not emitted in coverage-reports directory then also the below command gets executed resulted in a total-coverage.txt file, which then be analyzed by the getCoverage code to give coverage percentage as NaN. so the code serves as a check to avoid getting NaN coverage
`generateCovTxtCmd := exec.CommandContext(g.ctx, "go", "tool", "covdata", "textfmt", "-i="+os.Getenv("GOCOVERDIR"), "-o="+os.Getenv("GOCOVERDIR")+"/total-coverage.txt")` 

![image](https://github.com/keploy/keploy/assets/91385321/d60134f5-ddd0-4498-a9b9-00b15c4bde64)